tests/test_connector.rs

```diff
@@ -606,3 +611,259 @@ fn test_ft_transfer_call_fee_greater_than_amount() {
+        p.assert_success()
+    }
+}
+
+#[test]
+fn test_deposit_pausability() {
+    let (master_account, contract) = init(CUSTODIAN_ADDRESS);
+    let user_account = create_user_account(&master_account);
+
+    // 1st deposit call - should succeed
+    let promises = call_deposit_with_proof(&user_account, CONTRACT_ACC, PROOF_DATA_NEAR);
+    for p in promises.iter() {
+        assert!(p.is_some());
+        let p = p.as_ref().unwrap();
+        p.assert_success()
+    }
+
+    // Pause deposit
+    let res = call_set_paused_flags(&contract, CONTRACT_ACC, PAUSE_DEPOSIT);
+    res.assert_success();
+
+    // 2nd deposit call - should fail
+    // NB: We can use `PROOF_DATA_ETH` this will be just a different proof but the same deposit
+    // method which should be paused
+    let promises = call_deposit_with_proof(&user_account, CONTRACT_ACC, PROOF_DATA_ETH);
+    let num_promises = promises.len();
+    let p = promises[num_promises - 2].clone();
+    match p.unwrap().status() {
+        ExecutionStatus::Failure(_) => {}
+        _ => panic!(),
+    }
+
+    // Unpause all
+    let res = call_set_paused_flags(&contract, CONTRACT_ACC, UNPAUSE_ALL);
+    res.assert_success();
+
+    // 3rd deposit call - should succeed
+    let promises = call_deposit_with_proof(&user_account, CONTRACT_ACC, PROOF_DATA_ETH);
+    for p in promises.iter() {
+        assert!(p.is_some());
+        let p = p.as_ref().unwrap();
+        p.assert_success()
+    }
+}
+
+#[test]
+fn test_withdraw_near_pausability() {
+    let (master_account, contract) = init(CUSTODIAN_ADDRESS);
+    let user_account = create_user_account(&master_account);
+
+    call_deposit_near(&contract, CONTRACT_ACC);
+
+    let withdraw_amount = 100;
+    let recipient_addr = validate_eth_address(RECIPIENT_ETH_ADDRESS);
+    // 1st withdraw - should succeed
+    let res = user_account.call(
+        CONTRACT_ACC.to_string(),
+        "withdraw",
+        &WithdrawCallArgs {
+            recipient_address: recipient_addr,
+            amount: withdraw_amount,
+        }
+        .try_to_vec()
+        .unwrap(),
+        DEFAULT_GAS,
+        1,
+    );
+    res.assert_success();
+    let promises = res.promise_results();
+    assert!(promises.len() > 1);
+    for p in promises.iter() {
+        assert!(p.is_some());
+        let p = p.as_ref().unwrap();
+        p.assert_success()
+    }
+
+    // Pause withdraw
+    let res = call_set_paused_flags(&contract, CONTRACT_ACC, PAUSE_WITHDRAW);
+    res.assert_success();
+
+    // 2nd withdraw - should fail
+    let res = user_account.call(
+        CONTRACT_ACC.to_string(),
+        "withdraw",
+        &WithdrawCallArgs {
+            recipient_address: recipient_addr,
+            amount: withdraw_amount,
+        }
+        .try_to_vec()
+        .unwrap(),
+        DEFAULT_GAS,
+        1,
+    );
+    let promises = res.promise_results();
+    let p = promises[1].clone();
+    match p.unwrap().status() {
+        ExecutionStatus::Failure(_) => {}
```

Comment:
I wonder if it's worth trying to check specifically the error is raised because of the pause (e.g. by checking the error message), as opposed to some other reason. Just for the sake of completeness.

Reply:
Makes sense for me! It's worth adding the same for already existing tests there